pandas/tests/indexes/datetimes/test_constructors.py

```diff
@@ -1003,7 +1010,7 @@ def test_dti_ambiguous_matches_timestamp(self, tz, use_str, box_cls, request):
             mark = pytest.mark.xfail(reason="We implicitly get fold=0.")
             request.applymarker(mark)
 
-        with pytest.raises(pytz.AmbiguousTimeError, match=dtstr):
+        with pytest.raises(errors.AmbiguousTimeError, match=dtstr):
```

Comment:
I guess the comment above about not providing error compatability here would make testing more challenging. Maybe its something we can include as part of the `tz` fixture? May also be worth doing as a follow up instead of adding to this PR